doc/release/compatibility.rst

```diff
@@ -0,0 +1,150 @@
+..  _compatibility_guarantees:
+
+Compatibility guarantees
+========================
+
+Backwards compatibility is guaranteed between all versions in the same :term:`release series`.
+It is also appreciated but not guaranteed between different release series (major number changes).
+
+*   Pre-releases and releases of one release series are compatible in all
+    senses defined below (any release with any release).
+
+*   Pre-releases and releases of consequent series are compatible by data
+    layout, binary protocol, and replication protocol.
+
+*   No guarantees are given regarding compatibility between
+    pre-releases/releases of non-consequent release series if the opposite
+    is not stated in the :ref:`release notes <release_notes>`.
+
+*   No guarantees are given regarding compatibility between alpha/beta
+    versions and between alpha/beta and pre-release/release even within one series.
+
+..  _cg_data_layout:
+
+Binary data layout
+------------------
+
+Any newer release (its runtime) is backward compatible with any older one.
+It means the more recent release can work on top of data
+(``*.xlog``, ``*.snap``, ``*.vylog``, ``*.run``) from the older one.
+All functionality of the older release can work in this configuration.
+The same compatibility is maintained between :term:`release series` as well.
+
+An attempt to use a new feature results in one of the options:
+
+*   The attempt is successful.
+
+*   There is a meaningful error about the old data layout until the database schema upgrade.
+    It does not lead to a service outage or data corruption.
+    An instance can upgrade the data layout using the :ref:`box.schema.upgrade() <admin-upgrades>` call
+    to enable all new release features (when all instances of the replicaset are run on the same Tarantool version).
+
+..  _cg_binary_protocol:
+
+Binary protocol
+---------------
+
+The binary protocol evolves without breaking compatibility with old clients.
+All binary protocol requests operational in an older release should work on a newer one.
+They do not change the meaning.
+Responses have the same format, but mappings may contain fields not present in the older release.
+
+``net.box`` client of the older release is able to work
+with the newer one, except the features introduced in the newer release.
+``net.box`` client of the newer release is fully operational with the server
+running under the older one, except the features not implemented in the older release.
+
+..  _cg_replication_protocol:
+
+Replication protocol
+--------------------
+
+An instance run on a newer release may work as:
+
+*   upstream (master) of an instance with an older release
+
+*   downstream (replica) without a database schema upgrade.
+
+The database schema upgrade (``box.schema.upgrade()``) must be performed when all replicaset instances
+run on the same Tarantool version.
+The upgrade does not cause downtime if the application does not lean on internal schema representation.
```

Comment:
```suggestion
The upgrade does not cause downtime if the application does not lean on internal schema representation.
```
I suggest rephrasing it without double negation ("does not... if does not"). The current phrasing requires leaping through some logic hoops :)